boards/opentitan/src/tests/sip_hash.rs

```diff
@@ -66,35 +47,65 @@ impl<'a> hasher::Client<'a, 8> for SipHashTestCallback {
         assert_eq!(ret, 0x9ed5975598371f51);
 
         self.hash_done.set(true);
+        self.output_buf.replace(digest);
```

Comment:
You should replace input in `add_data_done()` as well

Reply:
How would you suggest this to be implemented? The cb here gets triggered as each slice is being added by the loop in the test? and since _data in the cb is a particular slice, would we want to save it into a separate buffer or track the cb count/index?

Reply:
Good question.

Could we track calls to `add_data_done()` and just replace that array?

Reply:
In the new commit, tried making a new buffer with each callback to hold the data, and at the end replacing the old one with it. However, this requires 2 static buffers to be allocated per instance (might be very inefficient). 

Thoughts on this?